test/DynamoCoreTests/MessageLogTests.cs

```diff
@@ -27,6 +57,36 @@ public void TestWarningMessageLog()
             ProtoCore.Runtime.WarningEntry warningEntry = runtimeCore.RuntimeStatus.Warnings.ElementAt(0);
             Assert.AreEqual(ProtoCore.Runtime.WarningID.Default, warningEntry.ID);
         }
-        
+
+        [Test]
+        public void MultiThreadWarnings()
+        {
+            Assert.DoesNotThrow(() =>
+            {
+                for (int i = 0; i < 3; i++)
+                    LogWarningMessageEvents.OnLogWarningMessage($"Test{i}");
+
+                waitHandle1 = 1;// make thread1 wait until waitHandle is 0
+
+                Thread thread1 = new Thread(AddWarning);
+                thread1.Start();
+
+                while (waitHandle2 == 0 && thread1.IsAlive)// wait for thread1 to reach the Warning iteration
+                {
+                    Thread.Sleep(100);
+                }
+
+                LogWarningMessageEvents.OnLogWarningMessage($"Test3");
```

Comment:
So at this point, a new warning is added on the main thread to the warnings list while at the same time `thread1` is iterating over the warnings list?

Reply:
Correct